src/CoreWCF.Kafka/src/CoreWCF/Channels/SaslMechanismHelper.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Confluent.Kafka;
+
+namespace CoreWCF.Channels;
+
+internal static class SaslMechanismHelper
+{
+    public static bool IsDefined(SaslMechanism value)
+    {
+        return value == SaslMechanism.Plain
+               || value == SaslMechanism.Gssapi
+               || value == SaslMechanism.ScramSha256
+               || value == SaslMechanism.ScramSha512
```

Comment:
I don't think we have code for GSSAPI , SCRAM... let's throw exception

Reply:
@birojnayak i kept this helper as is to allow one configure these sasl mechanisms at tranport binding level using a CustomBinding. 
However I introduced enum values of `KafkaCredentialType` that match all the possible SaslMechanism values so we can later expose these SaslMechanism values at the binding level. For the time being, at binding level, I'm throwing if one try to use Gssapi, Scram or OAuthBearer sasl mechanism. I think this should clarify what need to be done to add more SaslMechanism support at binding level.